examples/examples/balance_transfer_with_params.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019-2022 Parity Technologies (UK) Ltd.
+// This file is part of subxt.
+//
+// subxt is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// subxt is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with subxt.  If not, see <http://www.gnu.org/licenses/>.
+
+//! To run this example, a local polkadot node should be running. Example verified against polkadot 0.9.18-f6d6ab005d-aarch64-macos.
+//!
+//! E.g.
+//! ```bash
+//! curl "https://github.com/paritytech/polkadot/releases/download/v0.9.13/polkadot" --output /usr/local/bin/polkadot --location
+//! polkadot --dev --tmp
+//! ```
+
+use sp_keyring::AccountKeyring;
+use subxt::{
+    extrinsic::{
+        Era,
+        PlainTip,
+    },
+    ClientBuilder,
+    DefaultConfig,
+    PairSigner,
+    PolkadotExtrinsicParams,
+    PolkadotExtrinsicParamsBuilder as Params,
+};
+
+#[subxt::subxt(runtime_metadata_path = "examples/polkadot_metadata.scale")]
+pub mod polkadot {}
+
+#[async_std::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    env_logger::init();
+
+    let signer = PairSigner::new(AccountKeyring::Alice.pair());
+    let dest = AccountKeyring::Bob.to_account_id().into();
+
+    let api = ClientBuilder::new()
+        .build()
+        .await?
+        .to_runtime_api::<polkadot::RuntimeApi<DefaultConfig, PolkadotExtrinsicParams<DefaultConfig>>>();
+
+    // Configure the transaction tip and era:
+    let tx_params = Params::new()
+        .tip(PlainTip::new(20_000_000_000))
+        .era(Era::Immortal, *api.client.genesis());
+
+    // Send the transaction:
+    let hash = api
+        .tx()
+        .balances()
+        .transfer(dest, 123_456_789_012_345)
+        .sign_and_submit(&signer, tx_params)
```

Comment:
Not sure whether this is possible/desirable, but did you consider adding a builder style method onto `SubmittableExtrinsic` which allows setting the params, and would use the default if not specified e.g.

```
let hash = api
        .tx()
        .balances()
        .transfer(dest, 123_456_789_012_345)
        .params(tx_params)
        .sign_and_submit(&signer)
```
It would avoid the requirement of the `_default` method equivalents, on the other hand would be less explicit if adding params is the common case.

Reply:
Ooh interesting! I didn't consider that offhand, no.

One thing I like about having the `_default` and the param-taking methods is that it's up to the user whether the params actually do implement `Default` at all or not; the `Default` constraint is only on those `_default` methods now, so if for some reason somebody could not or did not want a default set of params for their chain, that'd work just fine.

Reply:
I say that but actually I think I left a Default clause somewhere I shouldn't have, so this is a good pointer to remove that!